packages/terra-alert/src/Alert.jsx

```diff
@@ -0,0 +1,280 @@
+import IconAlert from 'terra-icon/lib/icon/IconAlert';
+import IconError from 'terra-icon/lib/icon/IconError';
+import IconWarning from 'terra-icon/lib/icon/IconWarning';
+import IconRequired from 'terra-icon/lib/icon/IconRequired';
+import IconDiamond from 'terra-icon/lib/icon/IconDiamond';
+import IconInformation from 'terra-icon/lib/icon/IconInformation';
+import IconSuccess from 'terra-icon/lib/icon/IconSuccess';
+import 'terra-base/lib/baseStyles';
+import styles from './Alert.scss';
+
+const AlertTypes = {
+  ALERT: 'alert',
+  ERROR: 'error',
+  WARNING: 'warning',
+  ADVISORY: 'advisory',
+  INFO: 'info',
+  SUCCESS: 'success',
+  GAP_CHECKING: 'gap_checking',
+  OUTSIDE_RECORDS: 'outside_records',
+  CUSTOM: 'custom',
+};
+
+const propTypes = {
+  /**
+   * An action element to be added to the action section of the alert.
+   */
+  alertAction: PropTypes.element,
+  /**
+   * Child Nodes providing the message content for the alert. Can contain text and HTML.
+   */
+  children: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
+  /**
+   * The icon to be used for an alert of type custom.
+   */
+  customIcon: PropTypes.element,
+  /**
+   * The status bar color to be used for an alert of type custom.
+   */
+  customStatusColor: PropTypes.string,
+  /**
+   * Callback function triggered when Dismiss button is clicked. The presence of this prop will cause the Dismiss button to be included on the alert.
+   */
+  onDismiss: PropTypes.func,
+  /**
+   * The title for the alert which will be bolded.
+   */
+  title: PropTypes.string,
+  /**
+   * The type of alert to be rendered. One of Alert.Types.ALERT, Alert.Types.ERROR, Alert.Types.WARNING, Alert.Types.ADVISORY,
+   * Alert.Types.INFO, Alert.Types.SUCCESS, Alert.Types.GAP_CHECKING, Alert.Types.OUTSIDE_RECORDS, Alert.Types.CUSTOM.
+   * Use the Types attribute of the Alert component for access to these type strings.
+   */
+  type: PropTypes.oneOf([
+    AlertTypes.ALERT,
+    AlertTypes.ERROR,
+    AlertTypes.WARNING,
+    AlertTypes.ADVISORY,
+    AlertTypes.INFO,
+    AlertTypes.SUCCESS,
+    AlertTypes.GAP_CHECKING,
+    AlertTypes.OUTSIDE_RECORDS,
+    AlertTypes.CUSTOM,
+  ]),
+};
+
+const defaultProps = {
+  alertAction: null,
+  children: '',
+  customIcon: null,
+  customStatusColor: '',
+  onDismiss: null,
+  title: '',
+  type: AlertTypes.ALERT,
+};
+
+const getAlertClassName = (type) => {
+  switch (type) {
+    case AlertTypes.ALERT:
+      return styles.alertType;
+    case AlertTypes.ERROR:
+      return styles.errorType;
+    case AlertTypes.WARNING:
+      return styles.warningType;
+    case AlertTypes.ADVISORY:
+      return styles.advisoryType;
+    case AlertTypes.INFO:
+      return styles.infoType;
+    case AlertTypes.SUCCESS:
+      return styles.successType;
+    case AlertTypes.GAP_CHECKING:
+      return styles.gapCheckingType;
+    case AlertTypes.OUTSIDE_RECORDS:
+      return styles.outsideRecordsType;
+    case AlertTypes.CUSTOM:
```

Comment:
In the initial design discussions with @neilpfeiffer, he indicated that the icons for the standard types were not to be modifiable and could only be specified for the Custom Alert type.  He alluded to that in the spec: https://wiki.ucern.com/pages/viewpage.action?spaceKey=UserExperience&title=Terra+UI%3A+Approved+Patterns